File: java/hybrid/InfoHandler.java
public class InfoHandler {
    private final java.util.List<Integer> dataset;
    public InfoHandler(java.util.List<Integer> numbers) { this.dataset = numbers; }
    public Stats computeStats() {
        int total = 0, cnt = 0;
        int currentMax = Integer.MIN_VALUE, currentMin = Integer.MAX_VALUE;
        for (int n : dataset) {
            total += n; cnt++;
            if (n > currentMax) currentMax = n;
            if (n < currentMin) currentMin = n;
        }
        double avg = cnt==0 ? 0.0 : (double)total / cnt;
        return new Stats(avg, currentMax, currentMin);
    }
    public static record Stats(double average, Integer maximum, Integer minimum) {}
}

File: python_draft/src/javascript/hybrid/InfoHandler.js
class InfoHandler {
  constructor(numbers){ this.dataset = numbers; }
  computeStats(){
    let total=0, cnt=0;
    let currentMax=-Infinity, currentMin=Infinity;
    for(const n of this.dataset){
      total += n; cnt++;
      if(n>currentMax) currentMax = n;
      if(n<currentMin) currentMin = n;
    }
    return { average: cnt? total/cnt : 0, maximum: cnt? currentMax:null, minimum: cnt? currentMin:null };
  }
}
module.exports = { InfoHandler };
